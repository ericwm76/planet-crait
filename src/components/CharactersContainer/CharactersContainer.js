import React from 'react';
import './CharactersContainer.scss';
import Character from '../Character/Character';
import SideBar from '../SideBar/SideBar';

const CharactersContainer = ({ characters }) => {
  const characterCards = characters.map((character, i) => {
    return (
      <Character
        name={character.name}
        species={character.species}
        homeworldName={character.homeworldName}
        homeworldPop={character.homeworldPop}
        appearsIn={character.appearsIn}
        favorited={character.favorited}
        key={i}
      />
    )
  })

  return (
    <section className='character-container'>
      <SideBar />
      {characterCards}
    </section> 
  )
}

export default CharactersContainer;