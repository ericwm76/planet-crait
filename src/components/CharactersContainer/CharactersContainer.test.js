import React from 'react';
import { shallow } from 'enzyme';
import CharactersContainer from './CharactersContainer';

describe('CharactersContainer', () => {
    let wrapper;
    let mockCharacters = [
      {name: "Luke Skywalker", species: "Human", homeworldName: "Tatooine", homeworldPop: "200000", appearsIn: ["The Empire Strikes Back", "Return of the Jedi", "A New Hope", "Revenge of the Sith", "The Force Awakens"], favorited: false},
      {name: "C-3PO", species: "Droid", homeworldName: "Tatooine", homeworldPop: "200000", appearsIn: ["The Empire Strikes Back", "Return of the Jedi", "A New Hope", "The Phantom Menace", "Attack of the Clones", "Revenge of the Sith"], favorited: false}
      ];
    let mockName = 'Potatoe';
    let mockQuote = 'Do. Or Do not';
    let mockRank = 'Expert'

    beforeEach(() => {
      wrapper = shallow(<CharactersContainer characters={mockCharacters} name={mockName} quote={mockQuote} rank={mockRank}/>)
    })

    it('should match snapshot with all data passing in correctly', () => {
        expect(wrapper).toMatchSnapshot();
    })
})