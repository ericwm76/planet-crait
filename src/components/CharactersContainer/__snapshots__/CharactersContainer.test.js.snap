// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharactersContainer should match snapshot with all data passing in correctly 1`] = `
<section
  className="character-container"
>
  <SideBar />
  <ScrollText />
</section>
`;
